dora/core/server/master/src/main/java/alluxio/master/job/DoraLoadJob.java

```diff
@@ -133,88 +133,48 @@ public class DoraLoadJob extends AbstractJob<DoraLoadJob.DoraLoadTask> {
+   * @param loadMetadataOnly    if set to true, only metadata will be loaded without loading
+   *                            file data
    */
   public DoraLoadJob(
       String path,
       Optional<String> user, String jobId, OptionalLong bandwidth,
       boolean usePartialListing,
-      boolean verificationEnabled) {
+      boolean verificationEnabled,
+      boolean loadMetadataOnly) {
     super(user, jobId, new HashBasedWorkerAssignPolicy());
-    mPath = requireNonNull(path, "path is null");
+    mLoadRootAlluxioPath = requireNonNull(path, "path is null");
+    mLoadRootAlluxioUri = new AlluxioURI(mLoadRootAlluxioPath);
     Preconditions.checkArgument(
         !bandwidth.isPresent() || bandwidth.getAsLong() > 0,
         format("bandwidth should be greater than 0 if provided, get %s", bandwidth));
     mBandwidth = bandwidth;
     mUsePartialListing = usePartialListing;
     mVerificationEnabled = verificationEnabled;
-    FileSystem fs = FileSystem.Factory.create(Scheduler.getInstance().getFileSystemContext());
-    ListStatusPOptions listOptions = ListStatusPOptions
-        .newBuilder()
-        .setRecursive(true).build();
-    mFs = fs;
-    mFileIterator = Optional.of(new FileListFetcher(fs, mPath, listOptions, false));
-  }
-
-  private static class FileListFetcher implements Iterator<URIStatus> {
-    public static final int PREFETCH_SIZE = 1000;
-    private final FileSystem mFs;
-    private String mPath;
-    ListStatusPOptions mListOption;
-    public LinkedBlockingQueue<URIStatus> mFiles = new LinkedBlockingQueue<>();
-    private boolean mUsePartialListing = false;
-    private AtomicBoolean mIsDone = new AtomicBoolean(false);
-
-    public FileListFetcher(FileSystem fs, String path, ListStatusPOptions listOption,
-                           boolean usePartialListing) {
-      mPath = path;
-      mListOption = listOption;
-      mFs = fs;
-      mUsePartialListing = false; // default to false now
-    }
-
-    private int advance() {
-      if (mIsDone.get()) {
-        return 0;
-      }
-      try {
-        // TODO(lucy) paginate list a PREFETCH_SIZE here.
-        List<URIStatus> uriStatuses = mFs.listStatus(new AlluxioURI(mPath), mListOption);
-        if (uriStatuses == null) {
-          mIsDone.compareAndSet(false, true);
-          return 0;
-        }
-        uriStatuses.forEach(uriStatus -> mFiles.offer(uriStatus));
-        if (!mUsePartialListing) {
-          mIsDone.compareAndSet(false, true);
-        }
-        return uriStatuses.size();
-      } catch (IOException | AlluxioException e) {
-        throw AlluxioRuntimeException.from(e);
-      }
-    }
-
-    @Override
-    public boolean hasNext() {
-      while (!mIsDone.get() && mFiles.size() < PREFETCH_SIZE * 0.2) {
-        if (advance() <= 0) {
-          break;
-        }
-      }
-      return mFiles.peek() != null;
-    }
+    String ufsRoot = Configuration.getString(PropertyKey.DORA_CLIENT_UFS_ROOT);
```

Comment:
Do you think we can move this to client side and for server-side we only use ufs full path? So for mLoadRootAlluxioPath we can just use ufs full path as well